packages/compass-components/src/components/combobox-with-custom-option.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useMemo } from 'react';
+import { Combobox, ComboboxOption } from './combobox';
+import type {
+  ComboboxProps,
+  onChangeType,
+  SelectValueType,
+} from './combobox/Combobox.types';
+
+type ComboboxWithCustomOptionProps<T extends boolean> = ComboboxProps<T> & {
+  options: string[];
+  optionLabel?: string;
+};
+
+export const ComboboxWithCustomOption = <M extends boolean>({
+  onChange,
+  options,
+  optionLabel,
+  multiselect = false as M,
+  ...props
+}: ComboboxWithCustomOptionProps<M>) => {
+  const [search, setSearch] = useState('');
+  const comboboxOptions = useMemo(() => {
+    const _opts = options.map((option, index) => (
```

Comment:
I am missing some info here. Could you help me understand how the filtering of options based on search happen here? Is that managed within `Combobox`?

Reply:
Closing as this was discussed on call